File: backend/app/schemas/post.py
from pydantic import BaseModel, ConfigDict

from app.schemas.user import UserSimple


class PostCreate(BaseModel):
    title: str
    content: str
    category_id: int


class CategorySimple(BaseModel):
    id: int
    name: str

    model_config = ConfigDict(
        from_attributes=True
    )


class PostResponse(PostCreate):
    id: int
    user: UserSimple
    category: CategorySimple

    like_count: int = 0
    liked: bool = False

    model_config = ConfigDict(
        from_attributes=True
    )


class PaginatedPostsResponse(BaseModel):
    posts: list[PostResponse]
    total: int
    page: int
    limit: int

File: frontend/types/post.ts
export interface Category {
  id: number;
  name: string;
}


export interface Post {

  id: number;

  title: string;

  content: string;


  user_id: number;

  username: string;


  like_count: number;

  comment_count: number;


  liked_by_me: boolean;


  created_at: string;


  category?: Category;
}